src/main/java/org/json/XML.java

```diff
@@ -431,6 +431,9 @@ private static boolean parse(XMLTokener x, JSONObject context, String name, XMLP
                                             && jsonObject.opt(config.getcDataTagName()) != null) {
                                         context.accumulate(tagName, jsonObject.opt(config.getcDataTagName()));
                                     } else {
+                                        if (!config.shouldTrimWhiteSpace()) {
```

Comment:
Why was removeEmpty() placed here, and not anywhere else in the parse block? e.g. line 425

Reply:
This was where the extra whitespace between tags was inadvertently being added and so is removed immediately if the conditions are met